connections/stream-migration.md

```diff
@@ -43,18 +50,44 @@ skinparam sequenceMessageAlign center
 entity Initiator
 entity Responder
 
-note over Initiator, Responder: Assume at least 2 connections.
+' note over Initiator, Responder: Assume at least 2 connections.
+
+Initiator -> Responder: Open connection
+... <i>Establish both sides support multistream-select</i> ...
+
+Initiator -> Responder: Open multiplexed stream
+
+Initiator -> Responder: Send ""<stream-migration protocol id>/<stream-id>""\ne.g. ""streamMigration/1.0.0/A""
+Initiator -> Responder: Send ""<user-protocol-id>"" (as-in multistream select)
+
+alt Responder supports stream-migration
+  Initiator <- Responder: Echo back ""<user-protocol-id>"" (as-in multistream select)
+  note over Initiator
+   The initiator knows the peer supports stream-migration since it echoed back
+   the user protocol.
+  end note
 
-Initiator <-> Responder: <b>Connection 1; stream A</b>
-Initiator <-> Responder: <b>Connection 2</b>
+
+  note over Responder
+   The responder knows that this stream is called <b>stream A</b>
+  end note
+
+  ... <i>Continue negotiating another protocol</i> ...
+else Responder does <i>not</i> supports stream-migration
```

Comment:
Note: This should be _super rare_. We know from Identify which protocols the peer speaks, and we should only attempt the stream migration protocol if we know that the peer actually supports it. Otherwise we'll waste 1 RTT. We probably should a SHOULD somewhere in the text explaining this.